packages/gatsby-plugin-offline/src/sw-append.js

```diff
@@ -9,13 +9,19 @@ self.addEventListener(`message`, event => {
     event.waitUntil(
       caches.open(cacheName).then(cache =>
         Promise.all(
-          resources.map(resource =>
-            cache.add(resource).catch(e => {
-              // ignore TypeErrors - these are usually due to
-              // external resources which don't allow CORS
-              if (!(e instanceof TypeError)) throw e
-            })
-          )
+          resources.map(resource => {
+            let request
```

Comment:
Few questions here:

- Do we still want to throw on (certain?) errors? I suppose the promise will just reject?
- This seems weird that we've totally changed the approach, i.e. we're fetching everything now?
- This seems to fix things like google-analytics. Were you able to validate that?

Just asking these to make sure we're on the same page and I understand things!

Reply:
> Do we still want to throw on (certain?) errors? I suppose the promise will just reject?

If there are any odd errors then yes, the promise will reject the same way as before

> This seems weird that we've totally changed the approach, i.e. we're fetching everything now?

We were already fetching everything with the old code, just never with `no-cors`, so sometimes there would be console errors

> This seems to fix things like google-analytics. Were you able to validate that?

I haven't tested Google Analytics explicitly but yes, it should be fixed since no-cors means it can be cached. I've tested other external resources though to check it's okay.

Reply:
Maybe we validate with that one just to validate we've fixed the issue this is fixing?

Once that's done, I think this will be good to go and merge 🎉

Reply:
This is not a best practice and will quickly lead to improper caching of resources. 

From [Workbox's own documentation](https://developers.google.com/web/tools/workbox/guides/storage-quota#beware_of_opaque_responses):

> # Beware of opaque responses! 
> A common source of unexpectedly high quota usage is due to runtime caching of opaque responses, which is to say, cross-origin responses to requests made without CORS enabled.
>  
> Browsers automatically inflate the quota impact of those opaque responses as a security consideration. In Chrome, for instance, even an opaque response of a few kilobytes will end up contributing around 7 megabytes towards your quota usage.

This should probably be enabled on a case-by-case basis using the plugin's options or not at all. If the external domain allows CORS, the resource should flag it using the HTML `crossorigin` attribute.

Reply:
@wconnorwalsh awesome; thank you for the info! This change also (could have) illustrated exactly what you reference here:

> A common source of unexpectedly high quota usage is due to runtime caching of opaque responses

as we've seen some recent weirdness with gatsbyjs.org.

I'm mostly out today, would you be interested in reverting this change and making this opt-in rather than enabled by default? If not, we'll get to it as soon as we can!

Reply:
cc @wconnorwalsh @DSchau please note that only the changes to `sw-append.js` should be reverted - the others are unrelated!

Reply:
I've looked into this a bit more and going to create a PR shortly to revert this + fix a few more things.